File: frontend/lib/receitas-mensais.ts
// Faturamento mensal (RBT12) — manual + puxar da Receita.
import { apiFetch } from "./api";

export type MesReceita = {
  ano_mes: string;
  valor_interno: number;
  valor_externo: number;
  origem?: string | null;
};

export type ReceitasMensaisResposta = {
  empresa_id: number;
  competencia: string;
  meses: MesReceita[];
  rbt12: number;
  meses_preenchidos: number;
};

export type PuxarReceitaResposta = {
  empresa_id: number;
  competencia: string;
  meses: Array<{ ano_mes: string; valor_interno: number; valor_externo: number; encontrado: boolean }>;
  encontrados: number;
  total_meses: number;
  erros: string[];
  aviso?: string | null;
};

/** Lista os 12 meses anteriores à competência com valores (0 se vazio). */
export function listarReceitasMensais(empresaId: number, competencia: string) {
  return apiFetch<ReceitasMensaisResposta>(
    `/api/v1/empresas/${empresaId}/receitas-mensais?competencia=${competencia}`,
  );
}

/** Salva (upsert) os meses da grade manual. */
export function salvarReceitasMensais(empresaId: number, meses: MesReceita[]) {
  return apiFetch<{ salvos: number }>(
    `/api/v1/empresas/${empresaId}/receitas-mensais`,
    { method: "PUT", body: JSON.stringify({ meses }) },
  );
}

/** Puxa o faturamento dos 12 meses anteriores via Integra Contador. */
export function puxarReceitaDaReceita(empresaId: number, competencia: string) {
  return apiFetch<PuxarReceitaResposta>(
    `/api/v1/empresas/${empresaId}/receitas-mensais/puxar-receita?competencia=${competencia}`,
    { method: "POST" },
  );
}

File: backend/app/routes/receitas_mensais.py
"""Rotas de faturamento mensal (RBT12)."""
from fastapi import APIRouter, Body, Depends
from sqlalchemy.orm import Session

from app.database import get_db
from app.services.auth_service import get_current_user
from app.services.receita_mensal_service import ReceitaMensalService


router = APIRouter(
    prefix="/empresas/{empresa_id}/receitas-mensais",
    tags=["receitas-mensais"],
    dependencies=[Depends(get_current_user)],
)


@router.get("")
def listar(empresa_id: int, competencia: str, db: Session = Depends(get_db)) -> dict:
    """Lista os 12 meses anteriores a `competencia` (AAAAMM) com valores.

    Retorna {meses: [{ano_mes, valor_interno, valor_externo, origem}], rbt12,
    meses_preenchidos}. Meses sem dado vêm com valor 0.
    """
    return ReceitaMensalService(db).listar_para_competencia(empresa_id, competencia)


@router.put("")
def salvar(
    empresa_id: int,
    meses: list[dict] = Body(..., embed=True),
    db: Session = Depends(get_db),
) -> dict:
    """Salva (upsert) os meses informados na grade manual.

    Body: {"meses": [{ano_mes, valor_interno, valor_externo}, ...]}
    """
    return ReceitaMensalService(db).salvar_em_lote(empresa_id, meses, origem="manual")


@router.post("/puxar-receita")
def puxar_da_receita(
    empresa_id: int,
    competencia: str,
    db: Session = Depends(get_db),
) -> dict:
    """Puxa o faturamento dos 12 meses anteriores via Integra Contador
    (CONSDECLARACAO13 + CONSDECREC15). Best-effort — revise a grade depois.
    """
    return ReceitaMensalService(db).puxar_da_receita(empresa_id, competencia)
